authn/saml/src/main/java/se/swedenconnect/signservice/authn/saml/config/SamlAuthenticationHandlerFactory.java

```diff
@@ -165,6 +167,22 @@ else if (conf.getMetadataProviderRef() != null) {
         authnRequestGenerator, requestBinding);
   }
 
+  /**
+   * Asserts that a valid SAML type has been provided.
+   * 
+   * @param type the SAML type (if null, the default is assumed)
+   * @throws IllegalArgumentException for invalid types
+   */
+  protected void assertSamlType(@Nullable final String type) throws IllegalArgumentException {
+    if (type == null) {
+      return;
+    }
+    if (!SamlAuthenticationHandlerConfiguration.SAML_TYPE_SWEDEN_CONNECT.equals(type)
+        && !SamlAuthenticationHandlerConfiguration.SAML_TYPE_DEFAULT.equals(type)) {
+      throw new IllegalArgumentException("Unknown saml-type - " + type);
+    }
```

Comment:
Isn't this the typical case where the constructor should set a list of supported values instead of the set being programatically fixed?
Or is this never meant to support any other types than Sweden Connect?

Reply:
It's not that simple. The type tells how to instantiate other stuff, so we need to do this. Overriding classes introducing new types should override this method.